src/utils/upload/fileInspector.js

```diff
@@ -0,0 +1,66 @@
+import { Gunzip } from 'fflate';
+
+import techOptions from './fileUploadSpecifications';
+import readFileToBuffer from './readFileToBuffer';
+
+const Verdict = {
+  INVALID_NAME: -2,
+  INVALID_FORMAT: -1,
+  VALID_ZIPPED: 0,
+  VALID_UNZIPPED: 1,
+};
+
+const MATRIX_SIGNATURE = Buffer.from('%%MatrixMarket');
+const FEATURES_SIGNATURE = Buffer.from('ENS');
```

Comment:
Feature files can also start with `lnc` - https://this-is-biomage.slack.com/archives/C01CP46TJ0Z/p1627654709006400